src/main/kotlin/com/sourcegraph/cody/context/ui/EnhancedContextPanel.kt

```diff
@@ -60,8 +61,12 @@ class EnhancedContextPanel(private val project: Project) : JPanel() {
 
   init {
     layout = VerticalFlowLayout(VerticalFlowLayout.BOTTOM, 14, 0, true, false)
-    tree.expandRow(0)
 
-    add(toolbarPanel)
+    // Fix for https://github.com/sourcegraph/jetbrains/issues/344
```

Comment:
I think that fix is right.
Looks like `EnhancedContextPanel ` is created in `ChatPanel` constructor and `ChatPanel` in turn in the `CodyToolWindowContent`. 

`CodyToolWindowContent` is a IJ project-level service object, and its lifecycle is controlled by IJ. 
I found this info in IJ [documentation](https://plugins.jetbrains.com/docs/intellij/plugin-services.html#retrieving-a-service):

> Getting a service doesn't need a read action and can be performed from any thread. If a service is requested from several threads, it will be initialised in the first thread, and other threads will be blocked until it is fully initialised.

So you are probably right, it's a race. We need to be on a safe side there.